src/server/system_services/objects/quota.js

```diff
@@ -0,0 +1,198 @@
+        return this.size.value;
+    }
+
+    /**
+     * @returns size unit.
+     */
+    get_size_unit() {
+        return this.size.unit;
+    }
+
+    /**
+     * @returns quantity value.
+     */
+    get_quantity_value() {
+        return this.quantity.value;
+    }
+
+    /**
+     * @returns quantity unit.
+     */
+     get_quantity_unit() {
+        return this.quantity.unit;
+    }
+
+    /**
+     * 
+     * @returns calculated bigint size raw value of quota. default is 0.
+     */
+    get_quota_by_size() {
+        return this.size.raw_value;
+    }
+
+    /**
+     * 
+     * @returns calculated bigint quantity raw value of quota
+     */
+    get_quota_by_quantity() {
+        return this.quantity.raw_value;
+    }
+
+    /**
+     * 
+     * @returns - new quota config object without raw values
+     */
+    get_config() {
+        const quota_config = {};
+        if (this.size.value > 0) {
+            quota_config.size = _.omit(this.size, RAW_VALUE_FIELD_NAME);
+        }
+        if (this.quantity.value > 0) {
+            quota_config.quantity = _.omit(this.quantity, RAW_VALUE_FIELD_NAME);
+        }
+        return quota_config;
+    }
+
+
+  /**
+   * 
+   * @returns the percent of quantity quota used by the bucket. 
+   */
+    static get_bucket_quota_usages_percent(bucket, bucket_quota) {
+        var usage = {
+            size_used_percent: 0,
+            quantity_used_percent: 0
+        };
+        if (!bucket_quota) return usage;
+        const objects_size = bucket.storage_stats && size_utils.json_to_bigint(bucket.storage_stats.objects_size);
+        const quota_size = size_utils.json_to_bigint(bucket_quota.get_quota_by_size());
+        if (quota_size > 0) {
+            usage.size_used_percent = objects_size.multiply(100).divide(quota_size).valueOf();
+        }
+        const objects_count = bucket.storage_stats && size_utils.json_to_bigint(bucket.storage_stats.objects_count);
+        const quantity_quota = size_utils.json_to_bigint(bucket_quota.get_quota_by_quantity());
+        if (quantity_quota > 0) {
+            usage.quantity_used_percent = objects_count.multiply(100).divide(quantity_quota).valueOf();
+        }
+        return usage;
+    }
+
+    /**
+     * Create and add quota alerts to collection if usage greates than threshold 90%.
+     * @param {*} sysid sysid - system id
+     * @param {*} bucket - own bucket
+     * @param {*} quota - bucket quota
+     * @returns
+     */
+    static add_quota_alerts(sysid, bucket, quota, collection) {
+        const { size_used_percent, quantity_used_percent } = Quota.get_bucket_quota_usages_percent(bucket, quota);
+        const size_alert = Quota.get_quota_alert(sysid, bucket, size_used_percent, quota.get_quota_by_size(), 'size');
+        if (size_alert !== null) {
+            collection.push(size_alert);
+        }
+        const quantity_alert = Quota.get_quota_alert(sysid, bucket, quantity_used_percent, quota.get_quota_by_quantity(), 'quantity');
```

Comment:
For all those places that we call self functions, shouldn't it be `this` instead of `Quota.` ?

Reply:
These were static methods, but Danny already asked me to remove the static